core-3/src/main/java/scanning/engine/FourCylinderEngine.java

```diff
@@ -0,0 +1,16 @@
+package scanning.engine;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+@Component(value="oldSchoolStuff")
+@Qualifier("oldSchoolStuff")
```

Comment:
explain me, please, why do we need this qualifier?

Reply:
I use it because now I have 2 implementations of Engine, so I use it to establish the relation with the specified Car